packages/services/src/competition.service.ts

```diff
@@ -2292,6 +2292,21 @@ export class CompetitionService {
       );
     }
 
+    // Check VIP status early - VIPs bypass ALL requirements (stake, rank, etc.)
+    if (competition.vips && competition.vips.length > 0) {
```

Comment:
does this mean if a VIP is on the blocklist, the VIP will still pass? (since the blocklist is in validateParticipationRules)

Reply:
Also i point this out because in the PR desrciption
> 1. Blocklist (Hard Reject)
Highest priority - no bypasses
If agent in blocklist → Reject immediately
Error: "This agent is not permitted to join this competition"

but VIP's bypass the blocklist